src/bitcoin-cli.cpp

```diff
@@ -70,6 +71,7 @@ static void SetupCliArgs(ArgsManager& argsman)
     argsman.AddArg("-rpcport=<port>", strprintf("Connect to JSON-RPC on <port> (default: %u, testnet: %u, signet: %u, regtest: %u)", defaultBaseParams->RPCPort(), testnetBaseParams->RPCPort(), signetBaseParams->RPCPort(), regtestBaseParams->RPCPort()), ArgsManager::ALLOW_ANY | ArgsManager::NETWORK_ONLY, OptionsCategory::OPTIONS);
     argsman.AddArg("-rpcuser=<user>", "Username for JSON-RPC connections", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg("-rpcwait", "Wait for RPC server to start", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    argsman.AddArg("-rpcwaittimeout=<n>", strprintf("Timeout in seconds to wait for the RPC server to start, or 0 for no timeout. (default: %d)", DEFAULT_WAIT_CLIENT_TIMEOUT), ArgsManager::ALLOW_INT, OptionsCategory::OPTIONS);
```

Comment:
Judging from the PR description you likely thought about this, but could the timeout still be an argument in `-rpcwait` and only kick in if the value is greater than 1? That would avoid introducing another config option and as a bonus would be shorter to type. IDK what kind of timeout values API clients need.

Reply:
I'm quite happy to switch it around, but having `1` as a special value is kinda weird as well. I thought I'd go the most cautious route first and see which way wins in the review xD

Reply:
Fair. Is the real time ever that low anyway? Maybe counting -rpcwait=1 as one second would be fine.

Reply:
I don't expect many users to run into issues if we were to special case `-rpcwait=1`, but given the "creativity" of users I wouldn't put it past them to use `-rpcwait=42` or `-rpcwait=31337`. Some users seem to see "any value" as a personal challenge :smiley:

Reply:
I'd still be in favor of combining `rpcwait` and `rpcwaittimeout`, but if we prefer two arguments for this, it may be helpful to mention the relationship to `-rpcwait` in this description.